### PR TITLE
Fix auto scaling mode when calling tune method on trainer.

### LINKED ISSUE #7319: Auto scaling of batch size always runs on power mode.
## 🐛 Bug

When using `trainer.tune(...)` if `auto_scale_batch_size="binsearch"` the mode of autoscaling used will still be `power`.
 
## Please reproduce using the BoringModel

https://colab.research.google.com/drive/1e81qySuqSfOakcUxbgTVBKK23Iovwkdg?usp=sharing

### To Reproduce
Run the colab and use the instructions in the test to go through the trace.

### Expected behavior

`print(mode)` should be `binsearch` not `power`.

### Environment
Using the BoringModel env with  `pytorch_lightning.__version__=1.3.0rc2`.

### PR DESCRIPTION
**Pull request by ramonemiliani93**
## What does this PR do?
 Fixes #7319

Adding a test that should pass if an incorrect mode is used. The test also works to show that passing `binsearch` has no effect on the mode that is being run on the auto scaling. Not sure if the best way would be to update the `scale_batch_size_kwargs` in the `tune` method of the `Trainer`. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
Found the issue. In the Tuner class line 46 we need to insert something like this:

```python
            if isinstance(self.trainer.auto_scale_batch_size, str):
                scale_batch_size_kwargs.setdefault("mode", self.trainer.auto_scale_batch_size)
```

**Comment by carmocca** [comment]
> we need to insert something like this:

Exactly, sorry I broke this recently!

Pushed the fix

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/tuner/batch_size_scaling.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def scale_batch_size(
     elif mode == 'binsearch':
         new_size = _run_binsearch_scaling(trainer, model, new_size, batch_arg_name, max_trials)
     else:
-        raise ValueError('mode in method `scale_batch_size` can only be `power` or `binsearch')
+        raise ValueError('mode in method `scale_batch_size` can only be `power` or `binsearch`')
```
Comment (kaushikb11) [inline pytorch_lightning/tuner/batch_size_scaling.py:79]:
```suggestion
        raise ValueError('mode in method `scale_batch_size` could either be `power` or `binsearch`')
```

File: tests/tuner/test_scale_batch_size.py [hunk 2]
```diff
@@ -280,3 +280,27 @@ def test_scale_batch_size_no_trials(tmpdir):
...
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_val_batches=1,
+        limit_train_batches=1,
+        auto_scale_batch_size='ThisModeDoesNotExist',
+    )
+
+    with pytest.raises(ValueError, match='can only be `power` or `binsearch`'):
+        trainer.tune(model)
+    with pytest.raises(ValueError, match='can only be `power` or `binsearch`'):
```
Comment (kaushikb11) [inline tests/tuner/test_scale_batch_size.py:305]:
```suggestion
    with pytest.raises(ValueError, match='could either be `power` or `binsearch`'):
        trainer.tune(model)
    with pytest.raises(ValueError, match='could either be `power` or `binsearch`'):
```

**Review by kaushikb11 — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Review by awaelchli — APPROVED** [review]

**Pull request merged**